src/openforms/formio/tests/validation/helpers.py

```diff
@@ -24,8 +24,14 @@ def validate_formio_data(
     return is_valid, serializer.errors
 
 
-def extract_error(errors: ReturnDict, key: str, index: int = 0) -> ErrorDetail:
-    return errors[key][index]
+def extract_error(
+    errors: ReturnDict, key: str, index: int = 0, nested_field: str = ""
+) -> ErrorDetail | dict[str, list[ErrorDetail]] | None:
+    if isinstance(errors[key], list):
+        return errors[key][index]
+
+    if isinstance(errors[key], dict):
+        return errors[key][nested_field]["non_field_errors"][index]
```

Comment:
you can just use `extract_error(errors["addressNL"], "postcode")` instead of modifying this helper

This change introduces all kinds of type errors/warnings in existing code because the code doesn't check if it's `None`, `ErrorDetail` or a mapping of `ErrorDetail`s being returned and dealing with that in the tests creates so much bloat/noise